pilot/pkg/security/authz/matcher/header.go

```diff
@@ -23,14 +23,20 @@ import (
 )
 
 // HeaderMatcher converts a key, value string pair to a corresponding HeaderMatcher.
+// The wildcard "*" will be generated as ".+" instead of ".*".
 func HeaderMatcher(k, v string) *routepb.HeaderMatcher {
 	// We must check "*" first to make sure we'll generate a non empty value in the prefix/suffix case.
 	// Empty prefix/suffix value is invalid in HeaderMatcher.
 	if v == "*" {
 		return &routepb.HeaderMatcher{
 			Name: k,
-			HeaderMatchSpecifier: &routepb.HeaderMatcher_PresentMatch{
-				PresentMatch: true,
```

Comment:
If this is removed, how could we express if a header exists?

Reply:
Hi @hzxuzhonghu 
Thank you for reviewing my code.

Based on [Istio document](https://istio.io/latest/docs/reference/config/security/authorization-policy/)

> Presence match: “*” will match when value is not empty.

Present match means the value is not empty. Right now it checks if a header exists that's against the document and also the rest of code like [StringMatcher](https://github.com/istio/istio/blob/d0705cf0ed5591cc26c08001f3faab0a880aec48/pilot/pkg/security/authz/matcher/string.go#L50)

Reply:
Yes, i understand the context. This is the question most user will face, we need to consider carefully